src/components/forms/Search/index.tsx

```diff
@@ -0,0 +1,65 @@
+import { History, Location } from 'history';
+import { trimStart } from 'lodash';
+import querystring from 'querystring';
+import React, { useEffect, useState } from 'react';
+import { Button, Form, FormGroup, Input } from 'reactstrap';
+import { SEARCH } from '../../../configs/lang';
+
+/**
+ * Interface for handleSearchChange event handler
+ */
+export type Change = (event: React.ChangeEvent<HTMLInputElement>) => void;
+
+/**
+ * Interface for handleSubmit event handler
+ */
+export type Submit = (event: React.FormEvent<HTMLFormElement>) => void;
+
+/**
+ * Interface for SearchForm props
+ */
+export interface SearchFormProps {
+  history: History;
+  location: Location;
+  placeholder?: string;
```

Comment:
Can we make this not optional?  It seems we already have a desired default, which can be supplied to this component via **defaultProps**.  [Parse, don't validate](https://news.ycombinator.com/item?id=21476261).

Reply:
@kelvin-muchiri can we make this not optional? i.e. remove the `?`